templates/shibboleth/relying_party/_profile.erb

```diff
@@ -3,7 +3,7 @@
             <property name="<%= property['name'] %>">
                 <list>
                     <%- property['beans'].each do |bean| -%>
-                    <bean parent="<%= bean['parent'] %>"<%- if !bean['encryptAssertions'].nil? %> p:encryptAssertions="<%= bean['encryptAssertions'] %>"<% end %><% if !bean['encryptNameIDs'].nil? %> p:encryptNameIDs="<%= bean['encryptAssertions'] %>"<% end %> />
+                    <bean parent="<%= bean['parent'] %>"<%- if !bean['encryptAssertions'].nil? %> p:encryptAssertions="<%= bean['encryptAssertions'] %>"<% end %><% if !bean['encryptNameIDs'].nil? %> p:encryptNameIDs="<%= bean['encryptNameIDs'] %>"<% end %><% if !bean['nameIDFormatPrecedence'].nil? %> p:nameIDFormatPrecedence="<%= bean['nameIDFormatPrecedence'] %>"<% end %> />
                     <%- end -%>
```

Comment:
Since these properties match up to the tags, could you just loop and make a tag for each key/value pair instead of all the ifs. Then it would be more open to adding more options as well.